From 3c66e02f6248c37a38e70f13439d4e2acdf0f5ed Mon Sep 17 00:00:00 2001
From: SanCloud <github@sancloud.co.uk>
Date: Tue, 1 Mar 2016 13:02:35 +0000
Subject: [PATCH 5/5] Add SanCloud BBEnhanced setup

#rcn-ee: removed header.version=SE0A -> board_rev=BBEN, not u-boot upstreamable..
#for updated rev's add a $board_rev = SE0A; to include/configs/am335x_evm.h

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 board/ti/am335x/board.c      |  4 ++--
 board/ti/am335x/board.h      |  6 ++++++
 board/ti/am335x/mux.c        | 10 ++++++++--
 include/configs/am335x_evm.h |  2 ++
 4 files changed, 18 insertions(+), 4 deletions(-)

diff --git a/board/ti/am335x/board.c b/board/ti/am335x/board.c
index 49d95cf..bdeb8e4 100644
--- a/board/ti/am335x/board.c
+++ b/board/ti/am335x/board.c
@@ -645,7 +645,7 @@ int board_eth_init(bd_t *bis)
 	if (read_eeprom(&header) < 0)
 		puts("Could not get board ID.\n");
 
-	if (board_is_bone(&header) || board_is_bone_lt(&header) ||
+	if (board_is_bone(&header) || (board_is_bone_lt(&header) && !board_is_bone_lt_enhanced(&header)) ||
 	    board_is_idk(&header)) {
 		writel(MII_MODE_ENABLE, &cdev->miisel);
 		cpsw_slaves[0].phy_if = cpsw_slaves[1].phy_if =
@@ -675,7 +675,7 @@ int board_eth_init(bd_t *bis)
 #define AR8051_DEBUG_RGMII_CLK_DLY_REG	0x5
 #define AR8051_RGMII_TX_CLK_DLY		0x100
 
-	if (board_is_evm_sk(&header) || board_is_gp_evm(&header)) {
+	if (board_is_evm_sk(&header) || board_is_gp_evm(&header) || board_is_bone_lt_enhanced(&header)) {
 		const char *devname;
 		devname = miiphy_get_current_dev();
 
diff --git a/board/ti/am335x/board.h b/board/ti/am335x/board.h
index bc700d5..f083483 100644
--- a/board/ti/am335x/board.h
+++ b/board/ti/am335x/board.h
@@ -39,6 +39,12 @@ static inline int board_is_bone_lt(struct am335x_baseboard_id *header)
 	return !strncmp(header->name, "A335BNLT", HDR_NAME_LEN);
 }
 
+static inline int board_is_bone_lt_enhanced(struct am335x_baseboard_id *header)
+{
+	return (board_is_bone_lt(header) &&
+			strncmp("SE", header->version, 2) <= 0);
+}
+
 static inline int board_is_evm_sk(struct am335x_baseboard_id *header)
 {
 	return !strncmp("A335X_SK", header->name, HDR_NAME_LEN);
diff --git a/board/ti/am335x/mux.c b/board/ti/am335x/mux.c
index 79ed02f..e12d769 100644
--- a/board/ti/am335x/mux.c
+++ b/board/ti/am335x/mux.c
@@ -354,8 +354,14 @@ void enable_board_pin_mux(struct am335x_baseboard_id *header)
 		configure_module_pin_mux(rgmii1_pin_mux);
 		configure_module_pin_mux(mmc0_pin_mux_sk_evm);
 	} else if (board_is_bone_lt(header)) {
-		/* Beaglebone LT pinmux */
-		configure_module_pin_mux(mii1_pin_mux);
+		if(board_is_bone_lt_enhanced(header)) {
+			/* SanCloud Beaglebone LT Enhanced pinmux */
+			configure_module_pin_mux(rgmii1_pin_mux);
+		}
+		else {
+			/* Beaglebone LT pinmux */
+			configure_module_pin_mux(mii1_pin_mux);
+		}
 		configure_module_pin_mux(mmc0_pin_mux);
 #if defined(CONFIG_NAND) && defined(CONFIG_EMMC_BOOT)
 		configure_module_pin_mux(nand_pin_mux);
diff --git a/include/configs/am335x_evm.h b/include/configs/am335x_evm.h
index e3402c4..9643912 100644
--- a/include/configs/am335x_evm.h
+++ b/include/configs/am335x_evm.h
@@ -201,6 +201,8 @@
 				"setenv fdtfile am335x-bonegreen.dtb; setenv fdtbase am335x-bonegreen; " \
 			"elif test $board_rev = AIA0; then " \
 				"setenv fdtfile am335x-abbbi.dtb; setenv fdtbase am335x-abbbi; " \
+			"elif test $board_rev = SE0A; then " \
+				"setenv fdtfile am335x-boneenhanced.dtb; setenv fdtbase am335x-boneenhanced; " \
 			"else " \
 				"setenv fdtfile am335x-boneblack.dtb; setenv fdtbase am335x-boneblack; " \
 			"fi; " \
-- 
2.7.0

